src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/Kavita.kt

```diff
@@ -305,12 +338,15 @@ class Kavita(suffix: String = "") : ConfigurableSource, HttpSource() {
                         }
                     }
                 }
+                else -> isFilterOn = false
             }
         }
 
-        if (isFilterOn || query.isEmpty()) {
+        if (query.isEmpty()) {
+            isFilterOn = true
             return popularMangaRequest(page)
         } else {
+            isFilterOn = false
             return GET("$apiUrl/Library/search?queryString=$query", headers)
```

Comment:
I know you didn't touch this in this PR, but wouldn't this fail to parse queries with special characters, such as `?` or `&`? Consider this:
```suggestion
            val url = "$apiUrl/Library/search".toHttpUrl().newBuilder()
                .addQueryParameter("queryString", query)
            return GET(url.toString(), headers)
```

Reply:
I completely overlooked that. Thanks.

Reply:
you also need a `import okhttp3.HttpUrl.Companion.toHttpUrl`, wasn't sure how to add that to the suggestion